dali/operators/reader/nvdecoder/dynlink_nvcuvid.cc

```diff
@@ -1,163 +1,88 @@
-    printf("Failed to find required function \"%s\" in %s\n",       \
-            #name, __DriverLibName);                                \
-    return CUDA_ERROR_UNKNOWN;                                      \
+    if (!ret) {
+      printf("dlopen \"%s\" failed!\n", __DriverLibName);
+    }
   }
+  return ret;
+}
 
-#define GET_PROC_EX_V2(name, alias, required)                       \
-  alias = (t##name *)dlsym(driver_lib, STRINGIFY(name##_v2));        \
-  if (alias == NULL && required) {                                  \
-    printf("Failed to find required function \"%s\" in %s\n",       \
-            STRINGIFY(name##_v2), __DriverLibName);                 \
-    return CUDA_ERROR_UNKNOWN;                                      \
+void *LoadSymbol(const std::string &name) {
+  void *ret = nvcuvidDrvLib ? dlsym(nvcuvidDrvLib, name.c_str()) : nullptr;
+  if (!ret) {
+    printf("Failed to find required function \"%s\" in %s\n", name.c_str(), __DriverLibName);
   }
+  return ret;
+}
 
-#define CHECKED_CALL(call)          \
-  do {                              \
-    CUresult result = (call);       \
-    if (CUDA_SUCCESS != result) {   \
-      return result;                \
-    }                               \
-  } while(0)
-
-#define GET_PROC_REQUIRED(name) GET_PROC_EX(name,name,1)
-#define GET_PROC_OPTIONAL(name) GET_PROC_EX(name,name,0)
-#define GET_PROC(name)          GET_PROC_REQUIRED(name)
-#define GET_PROC_V2(name)       GET_PROC_EX_V2(name,name,1)
-
-CUresult cuvidInit(unsigned int Flags, DLLDRIVER &driver_lib)
-{
-  driver_lib = NULL;
-  CHECKED_CALL(LOAD_LIBRARY(&driver_lib));
-
-  // fetch all function pointers
-  GET_PROC(cuvidCreateVideoSource);
-  GET_PROC(cuvidCreateVideoSourceW);
-  GET_PROC(cuvidDestroyVideoSource);
-  GET_PROC(cuvidSetVideoSourceState);
-  GET_PROC(cuvidGetVideoSourceState);
-  GET_PROC(cuvidGetSourceVideoFormat);
-  GET_PROC(cuvidGetSourceAudioFormat);
-
-  GET_PROC(cuvidCreateVideoParser);
-  GET_PROC(cuvidParseVideoData);
-  GET_PROC(cuvidDestroyVideoParser);
-
-
-  GET_PROC(cuvidGetDecoderCaps);
-  GET_PROC(cuvidCreateDecoder);
-  GET_PROC(cuvidDestroyDecoder);
-  GET_PROC(cuvidDecodePicture);
-  GET_PROC_OPTIONAL(cuvidGetDecodeStatus);
-  GET_PROC_OPTIONAL(cuvidReconfigureDecoder);
-
-#if !defined(__CUVID_DEVPTR64) || defined(__CUVID_INTERNAL)
-  GET_PROC(cuvidMapVideoFrame);
-  GET_PROC(cuvidUnmapVideoFrame);
-#endif
-
-#if defined(_WIN64) || defined(__LP64__) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
-  GET_PROC(cuvidMapVideoFrame64);
-  GET_PROC(cuvidUnmapVideoFrame64);
-#endif
-  GET_PROC(cuvidCtxLockCreate);
-  GET_PROC(cuvidCtxLockDestroy);
-  GET_PROC(cuvidCtxLock);
-  GET_PROC(cuvidCtxUnlock);
-
-  return CUDA_SUCCESS;
 }
 
+// it is defined in the generated file
+typedef void *tLoadSymbol(const std::string &name);
+void NvcuvidSetSymbolLoader(tLoadSymbol loader_func);
+void Nvcuvid_2SetSymbolLoader(tLoadSymbol loader_func);
```

Comment:
why there's a `_2` in the name?

Reply:
There are two separated dynlink implementations, each have a separate `*SetSymbolLoader` function.